### ISSUE TITLE
Allowed memory size and large uploads

### ISSUE THREAD

**Issue by mikaelprag** [czim/laravel-paperclip#56]
Hi there, and thanks for the great plugin!

I'm uploading rather large files (2Gb+) via chunked uploads to Laravel (7). When the file is uploaded and done I'm trying to attach it to the paperclip model like so:

```
$file = new \Czim\FileHandling\Storage\File\SplFileInfoStorableFile();
$file->setData(new \SplFileInfo($path));
$file->setName("{$uniqueFileName}.{$extension}");
$postAsset->image = $file;
$postAsset->save();
```

I'm getting the error message:
>  Allowed memory size of 134217728 bytes exhausted (tried to allocate 2228990984 bytes) in <project path>/vendor/czim/file-handling/src/Storage/File/SplFileInfoStorableFile.php on line 60

So if I understand this correctly, to attach this file as it is now I need a "memory_limit" higher than the max file size?

Is there any other way of attaching the file that does not consume this amount of memory? Is it possible to do it in chunks, or append data to the paperclip model as the chunks come in or something similar?

**Comment by czim** [comment]
This is a design error on my part. The storable file interface has `content(): string` as part of its interface, which means that there's no flat-memory profile access to the content available.

The solution, I think, is to make the storable file interface return either a resource, or a Generator that returns chunks. The challenge there is keeping an interface that is broad enough, while also allowing for all implementations -- and it should work with, f.i. Imagine/GD. No doubt possible, but a bit of a challenge. Definitely a feature I want, so I'll put it on my to-do list.

**Comment by czim** [comment]
Just released version `3.2.0`, which relies on an update of the file-handling dependency. There may be some breaking changes there, depending on how much you've customized things.

Hopefully this solves your issue, as streaming file writes are now used where possible. This means that the SplFileInfo is used to read the file as a streaming resource, rather than attempt to read the entire contents at once.

Let me know if this solves the problem.

**Comment by mikaelprag** [comment]
It was just an example project at that point so I didn't change much. It works fine without a huge memory limit now, thank you!
